ios/RNSScreenStackAnimator.m

```diff
@@ -50,20 +50,23 @@ - (void)animateTransition:(id<UIViewControllerContextTransitioning>)transitionCo
   }
 
   if (screen != nil) {
-    if (screen.stackAnimation == RNSScreenStackAnimationSimplePush) {
-      [self animateSimplePushWithTransitionContext:transitionContext toVC:toViewController fromVC:fromViewController];
-    } else if (
-        screen.stackAnimation == RNSScreenStackAnimationFade || screen.stackAnimation == RNSScreenStackAnimationNone) {
-      [self animateFadeWithTransitionContext:transitionContext toVC:toViewController fromVC:fromViewController];
-    } else if (screen.stackAnimation == RNSScreenStackAnimationSlideFromBottom) {
-      [self animateSlideFromBottomWithTransitionContext:transitionContext
-                                                   toVC:toViewController
-                                                 fromVC:fromViewController];
-    } else if (screen.stackAnimation == RNSScreenStackAnimationFadeFromBottom) {
-      [self animateFadeFromBottomWithTransitionContext:transitionContext
-                                                  toVC:toViewController
-                                                fromVC:fromViewController];
+    if (screen.fullWidthGestureEnabled && transitionContext.isInteractive) {
+      // we are swiping with full width gesture
+      if (screen.customAnimationOnSwipe) {
+        [self animateTransitionWithStackAnimation:screen.stackAnimation
+                                transitionContext:transitionContext
+                                             toVC:toViewController
+                                           fromVC:fromViewController];
+      } else {
+        [self animateSimplePushWithTransitionContext:transitionContext toVC:toViewController fromVC:fromViewController];
+      }
+      return;
```

Comment:
I personally try to avoid early returns if there exists a more logical structure of the conditions. It appears like it'd suffice to put the remainig code in `else` block here and that'd result in the same behavior w/o a need of using `return` here